containerized.mk

```diff
@@ -39,19 +59,23 @@ ensure_sync_started:
 # to kill running containers (can't kill with ctrl-c)
 .PHONY: run
 run: ensure_image_exists
-	@ [ "$$DOCKER_SWARMKIT_DOCKER_RUN_CMD" ] || exit 1
-	@ DOCKER_RUN_COMMAND="docker run -t -v swarmkit-cache:${GOPATH}" \
+	[ "$$DOCKER_SWARMKIT_DOCKER_RUN_CMD" ] || exit 1
+	if [ "$$DOCKER_SWARMKIT_NOMOUNT" ]; then \
+			DOCKER_RUN_COMMAND="docker run -t"; \
+	else \
+		DOCKER_RUN_COMMAND="docker run -t -v swarmkit-cache:${GOPATH}" \
 		&& if [ "$$DOCKER_SWARMKIT_USE_DOCKER_SYNC" ]; then \
 			$(MAKE) ensure_sync_started && DOCKER_RUN_COMMAND+=" -v swarmkit-sync:${DOCKER_IMAGE_DIR}"; \
 		else \
 			DOCKER_RUN_COMMAND+=" -v ${ROOTDIR}:${DOCKER_IMAGE_DIR}"; \
 		fi \
 		&& if [ "$$DOCKER_SWARMKIT_USE_DELVE" ]; then \
 			DOCKER_RUN_COMMAND="DOCKER_SWARMKIT_DELVE_PORT=${DOCKER_SWARMKIT_DELVE_PORT} $$DOCKER_RUN_COMMAND" ; \
-			DOCKER_RUN_COMMAND+=" -p ${DOCKER_SWARMKIT_DELVE_PORT}:${DOCKER_SWARMKIT_DELVE_PORT} -e DOCKER_SWARMKIT_DELVE_PORT"; \
+		DOCKER_RUN_COMMAND+=" -p ${DOCKER_SWARMKIT_DELVE_PORT}:${DOCKER_SWARMKIT_DELVE_PORT} -e DOCKER_SWARMKIT_DELVE_PORT"; \
 			`# see https://github.com/derekparker/delve/issues/515#issuecomment-214911481'` ; \
 			DOCKER_RUN_COMMAND+=" --security-opt=seccomp:unconfined"; \
 		fi \
-		&& DOCKER_RUN_COMMAND+=" $$DOCKER_SWARMKIT_DOCKER_RUN_FLAGS $$DOCKER_SWARMKIT_EXTRA_RUN_FLAGS ${IMAGE_NAME} $$DOCKER_SWARMKIT_DOCKER_RUN_CMD" \
-		&& echo $$DOCKER_RUN_COMMAND \
-		&& eval $$DOCKER_RUN_COMMAND
+	fi \
+	&& DOCKER_RUN_COMMAND+=" $$DOCKER_SWARMKIT_DOCKER_RUN_FLAGS $$DOCKER_SWARMKIT_EXTRA_RUN_FLAGS ${IMAGE_NAME} $$DOCKER_SWARMKIT_DOCKER_RUN_CMD" \
+	&& echo $$DOCKER_RUN_COMMAND \
```

Comment:
I think you've made the whole thing more complicated than it needs to be, proposing a simple patch if you don't mind :)

Reply:
I think that's the only change you need here => https://github.com/docker/swarmkit/pull/2793/files

Reply:
Would it be an option to _reverse_ the default? i.e., don't bind by default, but have a `BIND_DIR` option to enable bind-mounting (similar to the moby repo)?

In case a bind-mount is used, the `COPY` step could also be skipped; see https://github.com/moby/moby/pull/38370